eslint/babel-eslint-parser/src/convert/convertAST.js

```diff
@@ -107,12 +107,7 @@ function convertProgramNode(ast) {
   if (ast.comments.length) {
     const lastComment = ast.comments[ast.comments.length - 1];
 
-    if (!ast.tokens.length) {
-      // if no tokens, the program starts at the end of the last comment
```

Comment:
This behavior doesn't match Espree's (it probably did at one point and has since changed).